### PR TITLE
serialContainer

### PR DESCRIPTION
**Pull request by berg0401**
Create MTC serial container and add it to a task container.

### CODE REVIEW

**Comment by berg0401** [comment]
@captain-yoshi

File: src/create_mtc_serial_container.cpp [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  MTC::ContainerBasePtr serialContainerBase = serialContainer;
+  setOutput(kPortSerialContainer, serialContainerBase);
+
+  return NodeStatus::SUCCESS;
+}
+
+BT::PortsList CreateMTCSerialContainer::providedPorts()
+{
+  return {
+    BT::OutputPort<MTC::ContainerBasePtr>(kPortSerialContainer, "MoveTo Serial Container"),
```
Comment (captain-yoshi) [inline src/create_mtc_serial_container.cpp]:
```suggestion
    BT::OutputPort<MTC::ContainerBasePtr>(kPortSerialContainer, "serial container"),
```

File: src/move_mtc_container_to_container.cpp [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      auto& container = *container_ptr;
+
+      container->add(std::move(unique_container_child));
+
+      return NodeStatus::SUCCESS;
+    }
+  }
+
+  return NodeStatus::FAILURE;
+  return NodeStatus::SUCCESS;
```
Comment (captain-yoshi) [inline src/move_mtc_container_to_container.cpp]:
```suggestion
```

File: src/move_mtc_container_to_container.cpp [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    }
+  }
+
+  if(auto any_container_ptr = getLockedPortContent(kPortParentContainer); unique_container_child)
+  {
+    if(auto* container_ptr = any_container_ptr->castPtr<MTC::ContainerBasePtr>())
+    {
+      auto& container = *container_ptr;
+
+      container->add(std::move(unique_container_child));
```
Comment (captain-yoshi) [inline src/move_mtc_container_to_container.cpp]:
```suggestion
  MTC::ContainerBase::pointer unique_child_container{ nullptr };
  if(auto any_container_ptr = getLockedPortContent(kPortChildContainer))
  {
    if(auto* container_ptr = any_container_ptr->castPtr<MTC::ContainerBasePtr>())
    {
      auto& child_container = *container_ptr;

      unique_child_container = sharedToUnique(child_container);
      any_container_ptr.assign(nullptr);  // set blackboard value to nullptr
    }
  }

  if(auto any_container_ptr = getLockedPortContent(kPortParentContainer); unique_container_child)
  {
    if(auto* container_ptr = any_container_ptr->castPtr<MTC::ContainerBasePtr>())
    {
      auto& parent_container = *container_ptr;

      parent_container->add(std::move(unique_child_container));
```

**Review by captain-yoshi — CHANGES_REQUESTED** [review]

**Comment by captain-yoshi** [comment]
I would also change the name `MoveMTCContainerToContainer` to `MoveMTCContainerToParentContainer`.

**Review by captain-yoshi — APPROVED** [review]

**Pull request merged**